agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -348,6 +431,30 @@ func (cs *CredentialSpecResource) clearCredentialSpec() {
 	}
 }
 
+func (cs *CredentialSpecResource) deleteTaskExecutionCredentialsRegKeys() error {
+	cs.lock.Lock()
+	defer cs.lock.Unlock()
+
+	return deleteTaskExecutionCredentialsRegKeysImpl(cs.taskARN)
+}
+
+func deleteTaskExecutionCredentialsRegKeys(taskARN string) error {
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, ecsCcgPluginRegistryKeyRoot, registry.ALL_ACCESS)
+	if err != nil {
+		// Early exit with success case, if the registry key doesn't exist then there are no task execution role creds to cleanup
```

Comment:
Will log error. 

If the customer touches the registry key and manually changes its permissions to deny the ECS agent (which runs as admin user), then we could potentially get a permissions issue while opening the registry key. Otherwise there should be no issues opening this key.